dist/index.js

```diff
@@ -13284,11 +13284,23 @@ async function main() {
         });
         version = latestRelease.data.tag_name.replace("v", "");
     }
+    const isVersionPreVersion1 = version.startsWith("0.");
     const targetPlatform = core.getInput("target-platform");
     let cachedPath = tc.find("sd", version);
     if (!cachedPath) {
-        const url = `https://github.com/chmln/sd/releases/download/v${version}/sd-v${version}-${targetPlatform}`;
-        const binPath = await tc.downloadTool(url);
+        let url = `https://github.com/chmln/sd/releases/download/v${version}/sd-v${version}-${targetPlatform}`;
+        // sd began adding a file extension starting on version v1.0.0
+        // Add the file extension unless the version is < v1.0.0
+        if (!isVersionPreVersion1) {
+            url += ".tar.gz";
+        }
+        let binPath = await tc.downloadTool(url);
+        // Because we downloaded a .tar.gz, we must extract it and then find the sd binary file within the extracted directory    
```

Comment:
I'll remove the extra whitespace.

```suggestion
        // Because we downloaded a .tar.gz, we must extract it and then find the sd binary file within the extracted directory
```